core/src/main/java/com/linecorp/armeria/server/EarlyResponseRoutingContext.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+final class EarlyResponseRoutingContext extends RoutingContextWrapper {
```

Comment:
How about introducing an enum like `RoutingStatus` instead of another implementation?

```
public enum RoutingStatus {
    OK,
    OPTIONS,
    INVALID_PATH,
}

public interface RoutingContext {
    ...
    RoutingStatus status();
}
```